Trees/BinarySearchTree.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef BINARY_SEARCH_TREE
+#define BINARY_SEARCH_TREE
+
+struct Node {
+    int value;
+    Node* left_child;
+    Node* right_child;
+};
+
+Node* get_node();
+
+class BinarySearchTree {
+private:
+    Node* root;
+
+    void traversal_inorder_helper(Node* tree);
+    void traversal_preorder_helper(Node* tree);
+    void traversal_postorder_helper(Node* tree);
```

Comment:
This is disallowed in C++. The best I can do is overload the same function name instead of naming it a 'helper'.